src/liballoc/vec.rs

```diff
@@ -2671,7 +2671,7 @@ impl<T> Iterator for IntoIter<T> {
                     // purposefully don't use 'ptr.offset' because for
                     // vectors with 0-size elements this would return the
                     // same pointer.
-                    self.ptr = arith_offset(self.ptr as *const i8, 1) as *mut T;
+                    self.ptr = arith_offset(self.ptr as *const T, 1) as *mut T;
```

Comment:
This is at least going to change behavior I think -- previously we'd increment the pointer by one byte but now this is a no-op as the size of T here is zero. Can you point out where else it's being similarly used?

Reply:
Oh, I didn't know using the type of `*const` affects the results, I use `*const T` here because `*mut T` is being used rather than `*mut i8`. I thought it meant the same.

Reply:
Pointers are valued in bytes, and the arith_offset function increments in size_of increments (i.e., equivalent to `self.ptr as usize + std::mem::size_of::<T>()*offset`).

Reply:
Ah, I didn't know about this. I will revert this part.